pulp_container/app/tasks/builder.py

```diff
@@ -124,6 +131,18 @@ def build_image_from_containerfile(
         working_directory = os.path.abspath(working_directory)
         context_path = os.path.join(working_directory, "context")
         os.makedirs(context_path, exist_ok=True)
+
+        if repo_version:
+            artifacts = {}
+            repo_version_artifacts = RepositoryVersion.objects.get(pk=repo_version).artifacts
+            files = FileContent.objects.filter(
+                digest__in=repo_version_artifacts.values("sha256")
+            ).values("_artifacts__pk", "relative_path")
+            if len(files) == 0:
+                raise ValidationError(_("No file found for the specified repository version."))
+            for file in files:
+                artifacts[str(file["_artifacts__pk"])] = file["relative_path"]
```

Comment:
It seems like we have the list of artifacts already available from `RepositoryVersion.objects.get(pk=repo_version).artifacts`. Have you considered iterating over this queryset instead of devising a new dictionary and then running another DB queries for getting the corresponding artifacts?

Reply:
Yes, but I could not find the `relative_path` (that will be used in the Containerfile) of each file in the *artifacts* queryset. Is there a better way to find it?

Reply:
Also, you may want to verify whether the artifacts are in place or not. A user could sync a repository with the on-demand policy and then use it as the build context for pulp-container. We should be careful about this.